### PR TITLE
CPP PYBIND Module for Tensor Map

### PR DESCRIPTION
**Pull request by DannyBruno**
Creates a cpp PYBIND Module for a tensor map. Includes additional changes for performance testing.

### CODE REVIEW

**Comment by tjruwase** [comment]
@DannyBruno, can you please share more details about this PR, including 
1. Motivation (link an issue or enhancement request).
2. What is accomplished by this PR.
3. Perf test results and environment.

Thanks!

**Comment by loadams** [comment]
Hi @DannyBruno - could you reply to the questions above?  Or if this PR is still needed/required?

File: csrc/map/tensor_map.cpp [hunk 1]
```diff
@@ -0,0 +1,49 @@
+#include <torch/extension.h>
```
Comment (loadams) [inline csrc/map/tensor_map.cpp:1]: For the new files, you'll need to add the license at the top of the file to pass the formatting check.

**Comment by loadams** [comment]
@DannyBruno - thoughts on completing or closing this PR, or would you like to see this merged but won't have time to work on it for a while?

**Comment by loadams** [comment]
Hi @DannyBruno - I'm closing this PR for now, due to it being fairly stale.  If you'd like to see this merged, let me know and we can re-open and get this merged?

**Comment by mrwyattii** [comment]
@loadams this was originally developed as a precursor to #2896 - but we found that this was no faster than storing cuda graph captures in a python dict. #2896 might also need to be closed. I think we have abandoned this work for the time being and plan to revisit it at a later time.

**Pull request closed without merging**